patches/CreatePBP.cmake

```diff
@@ -1,84 +1,150 @@
 # File defining macro outputting PSP-specific EBOOT.PBP out of passed executable target.
-# Copyright 2020 - Daniel 'dbeef' Zalega
-#
-# Args:
-# TARGET - defined by an add_executable call before calling create_pbp_file
-# TITLE - optional, string, target's name in PSP menu
-# ICON_PATH - optional, absolute path to .png file, 144x82
-# BACKGROUND_PATH - optional, absolute path to .png file, 480x272
-# PREVIEW_PATH - optional, absolute path to .png file, 480x272
 #
+# Copyright 2020 - Daniel 'dbeef' Zalega
+# Copyright 2021 - max_ishere
+
 cmake_minimum_required(VERSION 3.10)
 
 macro(create_pbp_file)
 
-    set(oneValueArgs TARGET TITLE ICON_PATH BACKGROUND_PATH PREVIEW_PATH)
-    cmake_parse_arguments("ARG" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
-
-    # As pack-pbp takes undefined arguments in form of "NULL" string,
-    # set each undefined macro variable to such value:
-    foreach(arg ${oneValueArgs})
-        if (NOT DEFINED ARG_${arg})
-            set(ARG_${arg} "NULL")
-        endif()
-    endforeach()
+  set(oneValueArgs
+    TARGET          # defined by an add_executable call before calling create_pbp_file
+    TITLE           # optional, string, target's name in PSP menu
+    ICON_PATH       # optional, absolute path to .png file, 144x82
+    BACKGROUND_PATH # optional, absolute path to .png file, 480x272
+    PREVIEW_PATH    # optional, absolute path to .png file, 480x272
+    )
+  set(options
+    BUILD_PRX # optional, generates and uses PRX file instead of ELF in EBOOT.PBP
+    ENC_PRX   # optional, replaces PRX file with encrypted version.
+    )
+  cmake_parse_arguments("ARG" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
-    if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
-        add_custom_command(
-            TARGET ${ARG_TARGET}
-            POST_BUILD COMMAND
-            "${STRIP}" "$<TARGET_FILE:${ARG_TARGET}>"
-            COMMENT "Stripping binary"
-        )
-    else()
-        add_custom_command(
-            TARGET ${ARG_TARGET}
-            POST_BUILD COMMAND
-            ${CMAKE_COMMAND} -E cmake_echo_color --cyan "Not stripping binary, build type is ${CMAKE_BUILD_TYPE}."
-        )
+  # As pack-pbp takes undefined arguments in form of "NULL" string,
+  # set each undefined macro variable to such value:
+  foreach(arg ${oneValueArgs})
+    if (NOT DEFINED ARG_${arg})
+      set(ARG_${arg} "NULL")
     endif()
+  endforeach()
+
+  if(NOT ${ARG_BUILD_PRX} AND ${ARG_ENC_PRX})
+    message(WARNING "You are asking to encrypt PRX that is not built by this macro.\n"
```

Comment:
Actually it makes sense to make it `FATAL_ERROR`

Reply:
From my experience, people ignore tons of warnings when generating their CMake projects, this one will get lost amidst them.

Reply:
Well, its just info. Its for people that wonder "Why no prx? I said `ENC_PRX`!"

But really its not an error. I can totally see someone putting every arg on a separate line and just commenting out BUILD_PRX line, without bothering about ENC.